client/state/terms/actions.js

```diff
@@ -1,15 +1,68 @@
 /**
+ * External dependencies
+ */
+import omit from 'lodash/omit';
+import uniqueId from 'lodash/uniqueId';
+
+/**
  * Internal dependencies
  */
 import wpcom from 'lib/wp';
 import {
+	TERMS_ADD_REQUEST,
+	TERMS_ADD_REQUEST_SUCCESS,
+	TERMS_ADD_REQUEST_FAILURE,
 	TERMS_RECEIVE,
 	TERMS_REQUEST,
 	TERMS_REQUEST_SUCCESS,
 	TERMS_REQUEST_FAILURE
 } from 'state/action-types';
 
 /**
+ * Returns an action thunk, dispatching progress of a request to add a new term
+ * the site and taxonomy.
+ *
+ * @param  {Number} siteId   Site ID
+ * @param  {String} taxonomy Taxonomy Slug
+ * @param  {Object} term     Object of new term attributes
+ * @return {Object}          Action object
+ */
+export function addTerm( siteId, taxonomy, term ) {
+	return async ( dispatch ) => {
+		const temporaryId = uniqueId( 'temporary' );
+
+		dispatch( {
+			type: TERMS_ADD_REQUEST,
+			temporaryId,
+			siteId,
+			taxonomy,
+			term
+		} );
+
+		try {
+			const data = await wpcom.site( siteId ).taxonomy( taxonomy ).term().add( term );
```

Comment:
Ah, I just love this syntax.

However, playing devil's advocate with myself. I started using `async` / `await` syntax in #6179 (in fact, I'm now noticing the `savePost` action still returns an `async` dispatch function... whoops!). The one thing I did not like is the appearance of the generated ES5 code, especially when trying to debug it by setting breakpoints. Promises aren't much better (then again, might be the fault of the `async` flag I left), but enough better that I wrote it that way. This wouldn't be an issue if we start using sourceMaps (slower builds, sometimes buggy behavior in Chrome dev tools).